server/src/routes/watchlist.js

```diff
@@ -1,17 +1,29 @@
 const express = require("express");
-const WatchLists = require('../db/mongo/models/watchlist');
+const HttpStatus = require("http-status-codes");
+const { validationResult } = require('express-validator/check');
+const {searchWatchlistByUsername} = require("../services/watchlist");
 
 const router = express.Router();
 
 router.get('/:username', async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        res.status(HttpStatus.BAD_REQUEST).json({ errors: errors.array() });
+        return;
+    }
+
+    let response;
+    let statusCode = HttpStatus.OK;
     const username  = req.params.username;
     try {
-        let watchlist = await WatchLists.find({ username: username}).exec();
-        res.send(watchlist);
+        response = await searchWatchlistByUsername(username);
```

Comment:
You can call it `getUserWatchlist` and afterwards you'll use my method straight away